controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -2080,7 +2080,7 @@ func (r *BareMetalHostReconciler) updateEventHandler(e event.UpdateEvent) bool {
 	}
 
 	// Discard updates that did not increase the resource Generation (such as on Status.LastUpdated), except for the finalizers or annotations
-	if reflect.DeepEqual(e.ObjectNew.GetFinalizers(), e.ObjectOld.GetFinalizers()) && reflect.DeepEqual(e.ObjectNew.GetAnnotations(), e.ObjectOld.GetAnnotations()) {
+	if reflect.DeepEqual(e.ObjectNew.GetFinalizers(), e.ObjectOld.GetFinalizers()) && reflect.DeepEqual(e.ObjectNew.GetAnnotations(), e.ObjectOld.GetAnnotations()) || oldBMH.Status.LastUpdated != newBMH.Status.LastUpdated {
 		return false
```

Comment:
This seems wrong, in the test below, we would want that event to come through because the finalizer is different from old/new, this prevents it because of the or. We can move this to a new line and isolate that.

Reply:
Added test and removed the logic on the line with the false condition.  Now this will be isolated from the event where the LastUpdated was modified but nothing else changed.